packages/apollo-gateway/src/datasources/__tests__/RemoteGraphQLDataSource.test.ts

```diff
@@ -115,6 +119,50 @@ describe('willSendRequest', () => {
   });
 });
 
+describe('didReceiveResponse', () => {
+  it('can accept and modify context', async () => {
+    interface MyContext {
+      surrogateKeys: string[];
+    }
+
+    const DataSource = new RemoteGraphQLDataSource({
+      url: 'https://api.example.com/foo',
+      async didReceiveResponse(response, request, context) {
+        // I'd prefer to use super.didReceiveResponse here but it doesn't seem to work.
+        const body = await RemoteGraphQLDataSource.prototype.didReceiveResponse.call(
+          this,
+          response,
+          request,
+          context,
+        );
+        const surrogateKeys = request.headers.get('surrogate-keys');
+        if (surrogateKeys) {
+          (context as any).surrogateKeys.push(...surrogateKeys.split(' '));
+        }
+        return body;
+      },
+    });
+
+    fetch.mockJSONResponseOnce({ data: { me: 'james' } });
```

Comment:
This test should fail.  You are not `james`.

:trollface:

Reply:
Alas, we can only aspire to be.